src/components/PrivateRoute.jsx

```diff
@@ -1,19 +1,54 @@
-import React from 'react';
+import '@reshuffle/code-transform/macro';
+import React, { useState, useEffect } from 'react';
 import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '@reshuffle/react-auth';
+import { checkIfValidDomain } from '../../backend/authBackend';
 
 /**
- * Standard React-router route that requires a userToken
+ * Standard React-router route that requires reshuffle identity
  */
-const PrivateRoute = ({ props, component: Component, userToken, ...rest }) => {
+const PrivateRoute = (props) => {
+  const { component: Component, ...rest } = props;
+  const { authenticated } = useAuth();
+  const [email, setEmail] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const value = await checkIfValidDomain();
+        setEmail(value);
+
+      } catch (err) {
+        console.error(err);
+      }
+    };
+    fetchData();
+  }, []);
+
+  if (authenticated === undefined) {
+    return null;
+  }
+
   return (
     <Route
       {...rest}
-      render={(routeProps) => {
-        if (userToken) {
-          return (<Component userToken={userToken} {...routeProps} {...props} />);
-        } else {
-          // if userToken is not found, redirect to auth page
-          return (<Redirect to='/auth'/>);
+      render={routeProps => {
+        if (authenticated && email) {
+          return <Component {...props} {...routeProps} />;
+        } else if (authenticated && !email) {
+          // if authenticated is false and the email is not the right domain, redirect to home page
+          return <Redirect to='/' />;
```

Comment:
Not sure I understand the user experience. Suppose I try to edit, get redirected to the auth page and mistakenly authenticate using my @gmail.com address. How do I reauthenticate back @reshuffle.com?